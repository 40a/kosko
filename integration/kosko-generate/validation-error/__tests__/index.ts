import execa from "execa";
import { dirname, join } from "path";
import { runCLI } from "../../../run";
import symlinkDir from "symlink-dir";
import pkgDir from "pkg-dir";

const testDir = dirname(__dirname);

let args: string[];
let result: execa.ExecaReturns;
let options: execa.Options;

beforeAll(async () => {
  const root = await pkgDir();
  await symlinkDir(
    join(root!, "packages", "env"),
    join(testDir, "node_modules", "@kosko", "env")
  );
});

beforeEach(async () => {
  result = await runCLI(args, {
    ...options,
    cwd: testDir
  });
});

describe("when validate is not set", () => {
  beforeAll(() => {
    args = ["generate"];
    options = { reject: false };
  });

  test("should return status code 1", () => {
    expect(result.code).toEqual(1);
  });

  test("should print the error", () => {
    expect(result.stderr).toMatchSnapshot();
  });
});

describe("when validate is true", () => {
  beforeAll(() => {
    args = ["generate"];
    options = { reject: false };
  });

  test("should return status code 1", () => {
    expect(result.code).toEqual(1);
  });

  test("should print the error", () => {
    expect(result.stderr).toMatchSnapshot();
  });
});

describe("when validate is false", () => {
  beforeAll(() => {
    args = ["generate", "--validate", "false"];
    options = {};
  });

  test("should return status code 0", () => {
    expect(result.code).toEqual(0);
  });

  test("should print the manifest", () => {
    expect(result.stdout).toMatchSnapshot();
  });
});
