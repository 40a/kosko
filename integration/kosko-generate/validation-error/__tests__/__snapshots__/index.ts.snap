// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when validate is false should print the manifest 1`] = `
"apiVersion: v1
kind: Pod
metadata:
  name: foo
spec: {}"
`;

exports[`when validate is not set should print the error 1`] = `
"error                        Validation failed <PROJECT_ROOT>/integration/kosko-generate/validation-error/components/foo.js
data.spec should have required property 'containers'
"
`;

exports[`when validate is true should print the error 1`] = `
"error                        Validation failed <PROJECT_ROOT>/integration/kosko-generate/validation-error/components/foo.js
data.spec should have required property 'containers'
"
`;
